L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -36,62 +41,58 @@ HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
     }
     else{
       int TC_wafer = TC_idV8.wafer();
-      int TC_12th = ( TC_idV8.cell() & kSplit12_ );
-      int TC_3rd = ( TC_idV8.cell() & kSplit3_ );
-      
       int thickness = triggerTools_.thicknessIndex(detid,true);
-      int TC_split = TC_12th;
-      if (stcSize_.at(thickness) == kSTCsize16_) TC_split = TC_3rd;
-      
+      int TC_split = ( TC_idV8.cell() & kSplit_[ stcSize_.at(thickness) ] );
+
       return TC_wafer<<kWafer_offset_ | TC_split;
     }
 
   }
 
   else if ( TC_id.det() == DetId::HGCalTrigger ){//V9
     
-    HGCalTriggerDetId TC_idV9(detid);
-
     if( triggerTools_.isScintillator(detid) ){
-      return TC_idV9.triggerCellU() << kWafer_offset_ | TC_idV9.triggerCellV(); //scintillator
+      HGCScintillatorDetId TC_idV9(detid);
+      return TC_idV9.ieta() << kWafer_offset_ | TC_idV9.iphi(); //scintillator
     }
     else {
+      HGCalTriggerDetId TC_idV9(detid);
+
       int TC_wafer = TC_idV9.waferU() << kWafer_offset_ | TC_idV9.waferV() ;
       int thickness = triggerTools_.thicknessIndex(detid);
 
-
-      int TC_cell = TC_idV9.triggerCellU() << kThree_ | TC_idV9.triggerCellV();
-
       int TC_12th = 0;
-      int TC_3rd = 1;
-
-      if ( TC_idV9.triggerCellU() > kThree_ || TC_idV9.triggerCellV() > kThree_ ){
+      int TC_3rd = 0;
+      int Uprime = 0;
+      int Vprime = 0;
+           
+      if ( TC_idV9.triggerCellU() > 3 || TC_idV9.triggerCellV() > 3 ){
         if ( TC_idV9.triggerCellU() <=  TC_idV9.triggerCellV() ){
 
-          TC_3rd = kTwo_;
-          TC_cell = (TC_idV9.triggerCellU()-TC_idV9.triggerCellV()+kThree_) << kThree_ | (TC_idV9.triggerCellV()-4);
-          TC_12th =  (TC_3rd << kTwo_) | ( TC_cell & kSplit12_v9 );
+          TC_3rd = 2;
+          Uprime = TC_idV9.triggerCellU()-TC_idV9.triggerCellV()+3;
+          Vprime = TC_idV9.triggerCellV()-4;
 
         }
         else{
 
-          TC_3rd = kThree_;
-          TC_cell = (TC_idV9.triggerCellU()-4) << kThree_ | (TC_idV9.triggerCellV()-TC_idV9.triggerCellU()+4);
-          TC_12th =  (TC_3rd << kTwo_) | ( TC_cell & kSplit12_v9 );
-
+          TC_3rd = 3;
+          Uprime = TC_idV9.triggerCellU()-4;
+          Vprime = TC_idV9.triggerCellV()-TC_idV9.triggerCellU()+4;
+          
         }
 
       }
       else{
 
-          TC_cell = (TC_idV9.triggerCellU()) << kThree_ | (TC_idV9.triggerCellV());
-          TC_12th =  (TC_3rd << kTwo_) | ( TC_cell & kSplit12_v9 );     
+          TC_3rd = 1;
+          Uprime = TC_idV9.triggerCellU();
+          Vprime = TC_idV9.triggerCellV();
 
       }
 
-
+      TC_12th =  (TC_3rd << 2) | ( (Uprime << 3 | Vprime) & kSplit12_v9 );
```

Comment:
It seems to me that some bits are overlapping. `TC_3rd ` needs 2 bits, `Uprime` and `Vprime` seem to be on 3 bits.
Then, shouldn't  `TC_3rd ` be shifted by 6 bits?